### PR TITLE
assorted fixes in pilot e2e tests

### PR DESCRIPTION
**Pull request by rshriram**
cc @nmittler

New option for Istio RBAC to enable/disable tests in pilot (default on)
Fix incorrect handling of per-service-auth enabled service D
Remove incorrect shadow/mirroring test. Added new test for mirroring
but skipping for now
Fix incorrect execution of auth enabled tests when auth is disabled

Signed-off-by: Shriram Rajagopalan <shriramr@vmware.com>

### CODE REVIEW

File: tests/e2e/tests/pilot/routing_test.go [hunk 1]
```diff
@@ -334,3 +325,37 @@ func TestRouteRedirectInjection(t *testing.T) {
 		}()
 	}
 }
+
+func TestRouteMirroring(t *testing.T) {
+	t.Skipf("Skipping %s", t.Name())
```
Comment (nmittler) [inline tests/e2e/tests/pilot/routing_test.go]: You're always skipping this test?

Comment (nmittler) [reply tests/e2e/tests/pilot/routing_test.go]: At least add a TODO?

File: tests/e2e/tests/pilot/tcp_test.go [hunk 2]
```diff
@@ -20,14 +20,16 @@ import (
...
-	dstPods := []string{"a", "b", "d"}
+	dstPods := []string{"a", "b"}
 	ports := []string{"90", "9090"}
 	if !tc.Kube.AuthEnabled {
 		// t is not behind proxy, so it cannot talk in Istio auth.
 		dstPods = append(dstPods, "t")
+	} else {
+		// Auth is enabled for d:9090 using per-service policy. We expect request
+		// from non-envoy client ("t") should fail all the time.
+		dstPods = append(dstPods, "d")
```
Comment (nmittler) [inline tests/e2e/tests/pilot/tcp_test.go]: Does this actually fix anything? It seems that the logic below handles things properly ... or am I missing something?

File: tests/e2e/tests/pilot/routing_test.go (hunk 1 above)
Comment (rshriram) [reply tests/e2e/tests/pilot/routing_test.go]: Yes.. It seems ths test always passes and I am not sure if I did it right.

File: tests/e2e/tests/pilot/tcp_test.go (hunk 2 above)
Comment (rshriram) [reply tests/e2e/tests/pilot/tcp_test.go]: It ensures that d is not even part of the test (sending/receiving requests)

Comment (ijsnellf) [reply tests/e2e/tests/pilot/tcp_test.go]: `d` is only part of the test when `AuthEnabled` is true...

**Pull request merged**